File: Sort/mergeSort.go
package main

import "fmt"

func mergesort(a []int, left, right int) {
	if left < right {
		mid := (left + right) / 2
		mergesort(a, left, mid)
		mergesort(a, mid+1, right)
		merge(a, left, mid, right)
	}
}
func merge(a []int, left, mid, right int) {
	i := left
	j := mid + 1
	k := left
	B := make([]int, right+1)
	for i <= mid && j <= right {

		if a[i] < a[j] {
			B[k] = a[i]
			i += 1
		} else {
			B[k] = a[j]
			j += 1
		}
		k = k + 1
	}
	for i <= mid {
		B[k] = a[i]
		i += 1
		k += 1
	}
	for j <= right {
		B[k] = a[j]
		j += 1
		k += 1
	}
	for x := left; x <= right; x++ {

		a[x] = B[x]
	}
}

func main() {
	items := []int{1, 2, 6, 8, 5, 4, 3, 7}
	fmt.Println(items)
	mergesort(items, 0, len(items)-1)
	fmt.Println(items)

}

File: Sort/mergeSort.py
def mergesort(a,left, right):
    if left<right:
        mid=(left+right)//2 
        mergesort(a,left,mid) 
        mergesort(a,mid+1,right)
        merge(a,left,mid,right) 
def merge(a,left,mid,right):
    i=left 
    j=mid+1 
    k=left 
    B=[0]*(right+1)
    while i<=mid and j<=right :
        if a[i]<a[j]:
            B[k]=a[i]
            i+=1 
        else:
            B[k]=a[j] 
            j+=1
        k=k+1 
    while i<=mid:
        B[k]=a[i]
        i+=1 
        k+=1 
    while j<=right:
        B[k]=a[j] 
        j+=1 
        k+=1 
    for x in range(left,right+1):
        a[x]=B[x]



A = [3, 5, 8, 9, 6, 2]
print('Original Array:',A)
mergesort(A,0,len(A)-1)
print('Sorted Array:',A)
